tidy3d/components/data/data_array.py

```diff
@@ -1052,6 +1052,36 @@ class ChargeDataArray(DataArray):
     _dims = ("n", "p")
 
 
+class IVCurveDataArray(DataArray):
+    """Semiconductor I-V curve data array.
+
+    Example
+    -------
+    >>> I = [0. 0, 1, 4]
+    >>> V = [-1, -0.5, 0, 0.5]
+    >>> td = IVCurveDataArray(data=I, coords={"Voltage (V)": V})
+    """
+
+    __slots__ = ()
+    _dims = ("Voltage (V)",)
+    _data_attrs = {"long_name": "Current (A)"}
+
+
+class CapacitanceCurveDataArray(DataArray):
```

Comment:
Likewise maybe feel like we need a more generic name for these "curves"? Ultimately these are parametric charge properties. But not sure what's the best naming as I guess it also depends on whether it's a static or dynamic data response.

Reply:
True. Similar situation as the IV curve. Not sure what's the best approach though. I can imagine a situation where we have the simulation run at different temperatures for which we'd have C(V,T) but also, once we do fully coupled heat-charge we'd also have T=T(X) though we could also have different reference temperatures. Let me ask Prash, but maybe a better name would be something like `CapacitanceDataArray` which we can later generalize to different dimensions.

Reply:
Yeah I like something along those lines! 

On the IV curves name, this is inherently a DC measurement and can't see us adding a dimension  here. I mean IVCurve is a good name, but I wonder for people building on top of it and separating data from plots, if something closer to DCVoltageCurrentData is more generic. Maybe also even more relevant for any other DC relationships eventually such as DCVoltageCapacitance for the dc shift of capacitances too (related to polarization nonlinearities if interested)

Reply:
Actually on this is on the same lines as the SPICE compatible data types. Ultimately a designer (I've been doing this myself) wants to extract an equivalent circuit model from the data classes in order to understand how the device behaves in a system - so we want to make sure our data types can comprehensively represent and be transformed to the electronic device physics that define these models.